snap/snapcraft.yaml

```diff
@@ -43,20 +61,75 @@ parts:
   intel-graphics:
     plugin: nil
     stage-packages:
-      - vainfo
+      - clinfo
+      - intel-gsc
+      - intel-level-zero-gpu
       - intel-media-va-driver-non-free
-      - libvpl2
-      - libvpl-dev
+      - intel-metrics-discovery
+      - intel-ocloc
+      - intel-ocloc-dev
+      - intel-opencl-icd
+      - libigc2
+      - libigc-dev
+      - libigc-tools
+      - libigdfcl2
+      - libigdfcl-dev
+      - libigdgmm12
+      - libigdgmm-dev
+      - libmfx1
+      - libmfx-gen1
+      - libmfx-gen-dev
+      - libva-dev
       - libva-drm2
       - libva-glx2
       - libva-wayland2
       - libva-x11-2
-      - libva-dev
-      - libmfx1
-      - libmfx-gen1
-      - libmfx-gen-dev
-      - libigdgmm12
-      - libigdgmm-dev
+      - libvpl2
+      - libvpl-dev
+      - libze1
+      - libze-dev
+      - libze-intel-gpu1
+      - libze-intel-gpu-dev
+      - ocl-icd-libopencl1
+      - vainfo
+
+  openvino-plugins:
+    source: https://github.com/intel/openvino-plugins-for-obs-studio.git
+    source-type: git
+    source-commit: v1.1-1-g128e
+    plugin: dump
+    after: [ 'obs-studio' ]
+    build-environment:
+      - OpenVINO_DIR: /snap/openvino-toolkit-2404/current/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/cmake/
+    build-packages:
+      - libopencv-dev
+      - cmake
+    build-snaps:
+      - openvino-toolkit-2404/latest/stable
+    override-build: |
+      set -eux
+      cd $CRAFT_PART_BUILD
+      cmake .
+      CORES="$(nproc)"
+      make -j$CORES
+      cp $CRAFT_PART_BUILD/intel64/Release/lib/*.so $CRAFT_PART_INSTALL/
+      mkdir $CRAFT_PART_INSTALL/{obs-smart-framing-ov,obs-face-mesh-ov,obs-backgroundconcealment-ov}
+      cp -R openvino-models/ $CRAFT_PART_INSTALL/
+    organize:
+      libobs-backgroundconcealment-ov.so: (component/openvino-plugins)/obs-plugins/
+      libobs-face-mesh-ov.so: (component/openvino-plugins)/obs-plugins/
+      libobs-smart-framing-ov.so: (component/openvino-plugins)/obs-plugins/
+      obs-smart-framing-ov: (component/openvino-plugins)/obs-plugins-data/obs-smart-framing-ov
+      obs-face-mesh-ov: (component/openvino-plugins)/obs-plugins-data/obs-face-mesh-ov
+      obs-backgroundconcealment-ov: (component/openvino-plugins)/obs-plugins-data/obs-backgroundconcealment-ov
+      openvino-models: (component/openvino-plugins)/obs-plugins-data/openvino-models
+      usr: (component/openvino-plugins)/usr
+    stage-packages:
+      - libopencv-imgproc406t64
+      - libopencv-imgcodecs406t64
+      - libopencv-core406t64
+    stage-snaps:
+      - openvino-toolkit-2404/latest/stable
```

Comment:
Would it be possible to add a cleanup part after all other parts to ensure that we're not shipping any redundant libraries or unnecessary files in the final snap? This will help reduce the snap size and avoid bundling unused content.